main_f1.c

```diff
@@ -15,16 +15,12 @@
 
 #include "bl.h"
 
-/*
- * Yes, the nonsense required to configure GPIOs with this
- * library is truly insane...
- */
-
-
-#ifdef INTERFACE_USART
-# define BOARD_INTERFACE_CONFIG		(void *)BOARD_USART
-#else
-# define BOARD_INTERFACE_CONFIG		NULL
+/* context passed to cinit */
+#if INTERFACE_USART
+# define BOARD_INTERFACE_CONFIG_USART	(void *)BOARD_USART
+#endif
+#if INTERFACE_USB
+# define BOARD_INTERFACE_CONFIG_USB  	NULL
 #endif
```

Comment:
Owing to my addition of the INTERFACE_USART in the hw_config.h file, this resulted in only teh USART interface being set.